File: Golang/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray_test.go
package RemoveDuplicatesFromSortedArray

import "testing"

func TestRemoveDuplicates(t *testing.T) {

	if removeDuplicates([]int{}) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates([]int{1, 2, 2}) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates([]int{1, 1, 1}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates([]int{1, 1, 2, 3, 4}) == 4 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/RemoveDuplicatesFromSortedArrayTest.py
from unittest import TestCase

from RemoveDuplicatesFromSortedArray import RemoveDuplicatesFromSortedArray


class TestRemoveDuplicatesFromSortedArray(TestCase):
    def test_removeDuplicates(self):
        r = RemoveDuplicatesFromSortedArray()

        self.assertTrue(r.removeDuplicates([]) == 0)

        self.assertTrue(r.removeDuplicates([1, 2, 2]) == 2)

        self.assertTrue(r.removeDuplicates([1, 1, 1]) == 1)

        self.assertTrue(r.removeDuplicates([1, 1, 2, 3, 4]) == 4)
